### PR TITLE
Add a fix where a panic in the sender go routine doesn't kill the ent…

### PR DESCRIPTION
**Pull request by schmidtw**
…ire machine.

#151

### CODE REVIEW

File: src/caduceus/mocks_test.go [hunk 1]
```diff
@@ -99,6 +100,11 @@ func (m *mockCounter) Add(delta float64) {
 }
 
 func (m *mockCounter) With(labelValues ...string) metrics.Counter {
+	for _, v := range labelValues {
+		if !utf8.ValidString(v) {
+			panic("not UTF-8")
+		}
+	}
 	args := m.Called(labelValues)
```
Comment (kristinaspring) [inline src/caduceus/mocks_test.go]: Can this line be before the labelValues check?

File: src/caduceus/mocks_test.go [hunk 2]
```diff
@@ -117,6 +123,11 @@ func (m *mockGauge) Set(value float64) {
 }
 
 func (m *mockGauge) With(labelValues ...string) metrics.Gauge {
+	for _, v := range labelValues {
+		if !utf8.ValidString(v) {
+			panic("not UTF-8")
+		}
+	}
 	args := m.Called(labelValues)
```
Comment (kristinaspring) [inline src/caduceus/mocks_test.go]: Can this line be before the labelValues check?

File: src/caduceus/mocks_test.go (hunk 1 above)
Comment (schmidtw) [reply src/caduceus/mocks_test.go]: yes - good catch

File: src/caduceus/mocks_test.go (hunk 2 above)
Comment (schmidtw) [reply src/caduceus/mocks_test.go]: yes - good catch

Comment (schmidtw) [reply src/caduceus/mocks_test.go]: I flipped it back so we don't need to add non UTF-8 values to the tests... we will clean this up when we do some pretty serious house cleaning on the code.

**Pull request merged**